Bio/SeqAnalysis/SeqDatabase.py

```diff
@@ -0,0 +1,277 @@
+                raise Exception("At least one element of your IDs list is not str.")
+
+        print("Number of unique IDs: {0}".format(len(clear_ids)))
+        print("Name of the current batch of files: {0}".format(analysis_dir))
+
+        print("------------------------------------Directory check------------------------------------------")
+
+        dirs = ['Downloads', 'Failed', 'Database']
+        for single_dir in dirs:
+            self.mk_dirs(single_dir)
+        self.mk_subdirs(analysis_dir)
+
+        obsolete = []
+        http_error = []
+
+        print("--------------------------------------Downloading--------------------------------------------")
+
+        urls = self.paths_and_urls(clear_ids, analysis_dir)
+
+        print("Downloading files. Please wait...")
+
+        ThreadPool(20).imap_unordered(self.fetch_url, urls)
+        start = timer()
+        for entry in urls:
+            self.fetch_url(entry, obsolete, http_error)
+
+        Entrez.email = ''
+        for idx in clear_ids:
+            if re.match(r'([A-Z]){2,3}\w+', idx):
+                self.download_ncbi(idx, analysis_dir, obsolete, http_error)
+
+        obsolete = list(dict.fromkeys(obsolete))
+
+        mypath = ('./Downloads/{0}'.format(analysis_dir))
+        only_files = [f for f in os.listdir(mypath) if isfile(join(mypath, f))]
+        files_no_ext = [".".join(f.split(".")[:-1]) for f in os.listdir(mypath) if os.path.isfile(join(mypath, f))]
+
+        total_number = len(only_files)
+        print("Download of " + str(total_number) + f" took {timer() - start} seconds")
+
+        print("-------------------------------------Saving outputs------------------------------------------")
+        print('Possibly obsolete entries were saved in Failed/{0}/Obsolete.txt'.format(analysis_dir))
+        print('Number of obsolete entries: {0}'.format(str(len(obsolete))))
+
+        with open('./Failed/{0}/Obsolete.txt'.format(analysis_dir), 'w+') as obs:
+            for item in obsolete:
+                obs.write(item + '\n')
+
+        print('Entries with possible download error were saved in Failed/{0}/Error.txt'.format(analysis_dir))
+        print('Number of download error entries: {0}'.format(str(len(http_error))))
+
+        with open('./Failed/{0}/Error.txt'.format(analysis_dir), 'w+') as fail:
+            for item in http_error:
+                fail.write(item + '\n')
+
+        print("Creating database. Please wait...")
+        for file in only_files:
+            with open('./Downloads/{0}/{1}'.format(analysis_dir, file)) as big_fasta:
+                one_fasta = big_fasta.read()
+                with open('./Database/{0}.fasta'.format(analysis_dir), 'a+') as data:
+                    data.write(one_fasta + '\n')
+
+        print("Your sequences were saved in Database/{0}.fasta".format(analysis_dir))
+        print("Done.")
+        self.downloaded_files = files_no_ext
+
+    @staticmethod
+    def read_list_of_ids(file: str) -> List:
```

Comment:
When using typing library, you may want to determine inner objects type:
`foo() -> List[str]:`